internal/commands/app/diff.go

```diff
@@ -125,12 +144,34 @@ func (cmd *CommandDiff) Handler(profile *user.Profile, ui terminal.UI, clients c
 }
 
 func (i *diffInputs) Resolve(profile *user.Profile, ui terminal.UI) error {
-	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory, true); err != nil {
+	searchPath := i.LocalPath
+	if searchPath == "" {
+		searchPath = profile.WorkingDirectory
+	}
+
+	app, err := local.LoadAppConfig(searchPath)
+	if err != nil {
 		return err
 	}
 
 	if i.LocalPath == "" {
-		i.LocalPath = profile.WorkingDirectory
+		if app.RootDir != "" {
+			i.LocalPath = app.RootDir
+		} else {
+			if err := ui.AskOne(&i.LocalPath, &survey.Input{Message: "App filepath (local)"}); err != nil {
+				return err
+			}
+
+			app, err = local.LoadAppConfig(i.LocalPath)
```

Comment:
⚠️  is this "go"?  I don't love the deeply-nested reassignment to `app` here, but tried to keep it idiomatic.  lmk if anyone sees any alternative approaches here!

Reply:
something like this might be cleaner:
```
	searchPath := i.LocalPath
	if searchPath == "" {
		searchPath = profile.WorkingDirectory
	}

	app, err := local.LoadAppConfig(searchPath)
	if err != nil {
		return err
	}

	if i.LocalPath == "" && app.RootDir == "" {
		if err := ui.AskOne(&i.LocalPath, &survey.Input{Message: "App filepath (local)"}); err != nil {
			return err
		}

		app, err = local.LoadAppConfig(i.LocalPath)
		if err != nil {
			return err
		}
	}
	i.LocalPath = app.RootDir

	if i.RemoteApp == "" {
		i.RemoteApp = app.Option()
	}

	return nil
```

Reply:
app isn't a pointer though so it would just default to an empty string

we have logic in `.Option()`:
```
func (a App) Option() string {
	if a.AppData == nil {
		return a.RootDir
	}
	if a.AppData.ID() != "" {
		return a.AppData.ID()
	}
	return a.AppData.Name()
}
```

Reply:
forgot I added nil checks in that function... go me